contracts/mainnet/connectors/liquity/main.sol

```diff
@@ -0,0 +1,426 @@
+    }
+
+    /**
+     * @dev Withdraw remaining ETH balance from user's redeemed Trove to their DSA
+     * @param setId Optional storage slot to store the ETH claimed
+     * @notice Claim remaining collateral from Trove
+    */
+    function claimCollateralFromRedemption(uint setId) external returns(string memory _eventName, bytes memory _eventParam) {
+        uint amount = collateralSurplus.getCollateral(address(this));
+        borrowerOperations.claimCollateral();
+        setUint(setId, amount);
+
+        _eventName = "LogClaimCollateralFromRedemption(address,uint256,uint256)";
+        _eventParam = abi.encode(msg.sender, amount, setId);
+    }
+    /* End: Trove */
+
+    /* Begin: Stability Pool */
+
+    /**
+     * @dev Deposit LUSD into Stability Pool
+     * @notice Deposit LUSD into Stability Pool
+     * @param amount Amount of LUSD to deposit into Stability Pool
+     * @param frontendTag Address of the frontend to make this deposit against (determines the kickback rate of rewards)
+     * @param getDepositId Optional storage slot to retrieve the LUSD from
+     * @param setEthGainId Optional storage slot to store any ETH gains in
+     * @param setLqtyGainId Optional storage slot to store any ETH gains in
+    */
+    function stabilityDeposit(
+        uint amount,
+        address frontendTag,
+        uint getDepositId,
+        uint setEthGainId,
+        uint setLqtyGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        amount = getUint(getDepositId, amount);
+        uint ethGain = stabilityPool.getDepositorETHGain(address(this));
+        uint lqtyGain = stabilityPool.getDepositorLQTYGain(address(this));
+        
+        stabilityPool.provideToSP(amount, frontendTag);
+        setUint(setEthGainId, ethGain);
+        setUint(setLqtyGainId, lqtyGain);
+
+        _eventName = "LogStabilityDeposit(address,uint256,address,uint256,uint256,uint256)";
+        _eventParam = abi.encode(msg.sender, amount, frontendTag, getDepositId, setEthGainId, setLqtyGainId);
+    }
+
+    /**
+     * @dev Withdraw user deposited LUSD from Stability Pool
+     * @notice Withdraw LUSD from Stability Pool
+     * @param amount Amount of LUSD to withdraw from Stability Pool
+     * @param setWithdrawId Optional storage slot to store the withdrawn LUSD
+     * @param setEthGainId Optional storage slot to store any ETH gains in
+     * @param setLqtyGainId Optional storage slot to store any ETH gains in
+    */
+    function stabilityWithdraw(
+        uint amount,
+        uint setWithdrawId,
+        uint setEthGainId,
+        uint setLqtyGainId
+    ) external returns (string memory _eventName, bytes memory _eventParam) {
+        stabilityPool.withdrawFromSP(amount);
+        uint ethGain = stabilityPool.getDepositorETHGain(address(this));
+        uint lqtyGain = stabilityPool.getDepositorLQTYGain(address(this));
```

Comment:
Again `getDepositorLQTYGain` returns slightly less than the LQTY that will be paid out